### DISCUSSION TITLE
Why is gateway showing ERR_UNEXPECTED_PROXY_AUTH error on brower?

### DISCUSSION THREAD

**Discussion by ChetanGhodke007** [ThreeMammals/Ocelot#1801] [Q&A]
407 (Proxy Authentication Required) status code

## Expected Behavior / New Feature
Gateway should return response from downstream api path 

## Actual Behavior / Motivation for New Feature
We have hosted 2 application on IIS (windows server), first downstream app opens on 1008 and ocelot gateway app opens on 1009 port. when called gateway upstream api url then gives error.

## Steps to Reproduce the Problem

  1. Create ocelot gateway app with following configuration
```json
  {
  "Routes": [
    {
      "DownstreamPathTemplate": "/api/values",
      "DownstreamScheme": "http",
      "DownstreamHttpMethod": "GET",
      "RouteIsCaseSensitive": false,
      "DownstreamHostAndPorts": [
        {
          "Host": "localhost",
          "Port": 1008
        }
      ],
      "UpstreamPathTemplate": "/gateway/values",
      "UpstreamHttpMethod": [ "GET" ]
    }
  ],

  "GlobalConfiguration": {
    "BaseUrl": "http://localhost:1009/"

  }
}
```
2. create simple api with get method.
3. call gateway url using `http://localhost:1009/gateway/values`

**While checking windows application logs below error logged:**

Category: Ocelot.Requester.Middleware.HttpRequesterMiddleware
EventId: 0
SpanId: bc5b99469ceadbf8
TraceId: dfe677d0d8dfb6548b6865513454e2db
ParentId: 0000000000000000
RequestId: 80000040-0004-f600-b63f-84710c7967bb
RequestPath: /gateway/values

requestId: 80000040-0004-f600-b63f-84710c7967bb, previousRequestId: no previous request id, message: **407 (Proxy Authentication Required) status code,** request uri: http://localhost:1008/api/values

## Specifications
  - Version: Ocelot 20.0.0
  - Platform: Windows
  - Subsystem: Asp.Net Core 7.0

**Comment by raman-m** [comment]
Welcome to Ocelot world! 🐯 

Why is your GitHub profile empty? Are you a student?

Please, fork Ocelot repository!... to be able to debug solution and contribute, if you have an intention to contribute, for sure

**Comment by raman-m** [comment]
@ks1990cn Hi Tanmay!
Please, help this guy!

Seems auth issue is related to `[Authorize]` filter attribute in a ASP.NET controller of his solution 😉 
But he must show whole solution to get an answer from us...

**Comment by ks1990cn** [comment]
@ChetanGhodke007 Please watch this tutorial, I think you want to do same. Just trying to pass JWT to reach downstream, may miss anything in configuration.
https://www.youtube.com/watch?v=P2osfctiHAc

**Comment by ChetanGhodke007** [comment]
@raman-m and @ks1990cn. Thank you for the reply. 
There is no any issue with Ocelot and its configuration when I ran both applications on local system. When I configured apps on IIS server then it gives 407 error. Note: I'm using .Net Core 7.0 for both application and not used [Authorize] filter attribute and JWT.

**Reply by ks1990cn** [reply]
Hahahaha! He is doing beyond our imagination @raman-m 🤣🤣🤣🤣

**Reply by raman-m** [reply]
[IIS Gotchas](https://ocelot.readthedocs.io/en/latest/introduction/gotchas.html#iis)
We don't recommend to host Ocelot app in IIS... Use self-hosted (Kestrel web server) and Docker instead.

**Reply by raman-m** [reply]
@ChetanGhodke007  Good that you've understood root cause, and closed this discussion.
Good luck in Ocelot development!

**Comment by ChetanGhodke007** [comment]
I have figure out issue. Used hostname instead of IP or localhost. Thanks!

**Reply by raman-m** [reply]
😆 That's so funny thread! 

@ks1990cn @ggnaegi 
Enjoy the moment of truth! 😋

**Reply by raman-m** [reply]
Amazing!  Learn more about DNS! 😉
